official/transformer/v2/transformer_main.py

```diff
@@ -125,24 +125,38 @@ def train(self):
         map_data_fn, num_parallel_calls=params["num_parallel_calls"])
 
     init_epoch = flags_obj.init_epoch or 0
-    init_steps = init_epoch * flags_obj.steps_per_epoch
-    callbacks = self._create_callbacks(cur_log_dir, init_steps, params)
+    init_steps = init_epoch * flags_obj.train_steps
+    callbacks = self._create_callbacks(flags_obj.model_dir, init_steps, params)
 
     history = model.fit(
         train_ds,
-        initial_epoch=init_epoch,
         epochs=flags_obj.train_epochs,
-        steps_per_epoch=flags_obj.steps_per_epoch,
+        steps_per_epoch=flags_obj.train_steps,
         validation_data=valid_ds,
         validation_steps=flags_obj.validation_steps,
-        callbacks=callbacks)
+        validation_freq=flags_obj.epochs_between_evals,
+        callbacks=callbacks,
+        verbose=2)
     tf.compat.v1.logging.info("\nTrain history: {}".format(history.history))
 
-    save_weight_path = os.path.join(cur_log_dir, "saves-model-weights.hdf5")
-    save_model_path = os.path.join(cur_log_dir, "saves-model.hdf5")
+    stats = misc.build_stats(history, callbacks)
+
+    if flags_obj.bleu_source and flags_obj.bleu_ref:
+      uncased_score, cased_score = evaluate_and_log_bleu(model,
```

Comment:
oh, this might be why you saw that error. the model in transformer is different for training vs eval/predict.
you need to create another model here: `model = transformer.create_model(params, is_train=False)`

Although, that makes me think why validation during training works with the same model.. @lintian06 do you know?

Reply:
Validation doesn't actually work, well it works but it does not give the values we want.  I guess I don't get why this works just fine with Resnet and we don't set training=False.  I will dig in more today.

Reply:
if you look inside the `create_model` function, you will see that an different model object is created for training vs eval/predict. it's to do with how custom losses are handled during training - which is not needed in eval so the eval model is slightly simpler. in resnet it's the same model in both cases.

Reply:
Fixed

Reply:
Validation is only to give a number. As it also feeds ground-truth labels, it is not a correct way to get BLEU score by bringing in extra knowledge. is_train=False will build another model which is totally agnostic to labels.